lib/srv/db/access_test.go

```diff
@@ -99,83 +99,130 @@ func TestMain(m *testing.M) {
+			allowDbLabels: wildCardDBLabels,
+			dbName:        "postgres",
+			dbUser:        "postgres",
 		},
 		{
-			desc:         "has access to nothing",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{},
-			allowDbUsers: []string{},
-			dbName:       "postgres",
-			dbUser:       "postgres",
-			err:          "access to db denied",
+			desc:          "has access to nothing",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{},
+			allowDbUsers:  []string{},
+			allowDbLabels: wildCardDBLabels,
+			dbName:        "postgres",
+			dbUser:        "postgres",
+			err:           "access to db denied",
 		},
 		{
-			desc:         "no access to databases",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{},
-			allowDbUsers: []string{types.Wildcard},
-			dbName:       "postgres",
-			dbUser:       "postgres",
-			err:          "access to db denied",
+			desc:          "no access to databases",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{},
+			allowDbUsers:  []string{types.Wildcard},
+			allowDbLabels: wildCardDBLabels,
+			dbName:        "postgres",
+			dbUser:        "postgres",
+			err:           "access to db denied",
 		},
 		{
-			desc:         "no access to users",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{types.Wildcard},
-			allowDbUsers: []string{},
-			dbName:       "postgres",
-			dbUser:       "postgres",
-			err:          "access to db denied",
+			desc:          "no access to users",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{types.Wildcard},
+			allowDbUsers:  []string{},
+			allowDbLabels: wildCardDBLabels,
+			dbName:        "postgres",
+			dbUser:        "postgres",
+			err:           "access to db denied",
 		},
 		{
-			desc:         "access allowed to specific user/database",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{"metrics"},
-			allowDbUsers: []string{"alice"},
-			dbName:       "metrics",
-			dbUser:       "alice",
+			desc:          "access allowed to specific user/database",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{"metrics"},
+			allowDbUsers:  []string{"alice"},
+			allowDbLabels: wildCardDBLabels,
+			dbName:        "metrics",
+			dbUser:        "alice",
 		},
 		{
-			desc:         "access denied to specific user/database",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{"metrics"},
-			allowDbUsers: []string{"alice"},
-			dbName:       "postgres",
-			dbUser:       "postgres",
-			err:          "access to db denied",
+			desc:          "access allowed to specific user/database by static label",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{"metrics"},
+			allowDbUsers:  []string{"alice"},
+			allowDbLabels: staticDBLabels,
```

Comment:
I'm finding it difficult to reason about this table. What is it about each entry that makes it work or not? Could we add a short comment to the new entries at least?

Reply:
added clarifying comments to the new test cases: [01c26d1](https://github.com/gravitational/teleport/pull/29215/commits/01c26d1eea4c5bd04074b89a0ba07ff8ce858d7e)